drivers/dai/sof/ssp/ssp.h

```diff
@@ -0,0 +1,375 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
```

Comment:
why do we need to have a path with 'sof'. Shouldn't this driver be generic to Zephyr?

Reply:
yeah good question. I was starting to add dmic driver and flat directory was becoming really messy. I tried to browse the zephyr driver tree and almost all directories are flat and the drivers are code wise smaller than this. Some of the drivers I'm trying to port are quite big and have auxiliary headers. I could try to lump all of that into 1 .c and 1 .h but that would make those files huge.

So yes, this should be (and is) generic driver, just I want to make it clear what is part of what when more drivers are appearing. Maybe remove sof and just add "ssp" dir?

Reply:
@juimonen I dont like flat directories, anyhow I was suggesting that `sof` in the name of the directory is not OK as we are trying to write application agnostic drivers.

My first preference for SSP driver should be something like: `drivers/i2s/intel/ssp.c`. If we decide that `i2s` interface is not what we want then we can go with `drivers/dai/intel/ssp.c`

Reply:
ack.